### PR TITLE
Clang tidy

### PR DESCRIPTION
**Pull request by thetic**
Most of these issues are pretty innocuous, but I did find a couple places that constructors/destructors were relying on dynamic dispatch.

### CODE REVIEW

**Comment by thetic** [comment]
It looks to me like the CodeQL workflow is running clang-tidy under the hood, but isn't honoring the `// NOLINT` comments. What do you think about deleting it? The inline comments in PRs are really nice, but I don't think you can run CodeQL locally.

@offa I think you added CodeQL. What do you think?

**Comment by thetic** [comment]
> [Coverage Status](https://coveralls.io/builds/54589269)
> 
> Coverage decreased (-0.0001%) to 99.625% when pulling **[d30c7b4](https://github.com/cpputest/cpputest/commit/d30c7b4f2b739e9fa003d10863c44961b1affe48) on thetic:clang-tidy** into **[5a715cf](https://github.com/cpputest/cpputest/commit/5a715cfa673a2674b4e8d80e40ae6d979b00e1b8) on cpputest:master**.

I believe coverage went down because the `StringFromOrdinalNumber` implementation got smaller.

**Comment by offa** [comment]
> What do you think about deleting it?

Just to not confuse this: Delete what? The inline suppression?

**Comment by offa** [comment]
I don't think they are same. CodeQL does more than static code analysis, eg. it contains queries for known security issues and variant analysis. Since it's implemented around queries I wouldn't expect it to call clang-tidy. As far as I know the former are implemented by Github or security researchers and do not use third party rules directly (I may be wrong here though).

While it may report the same result as clang-tidy here, both are different tools. Indeed, the missing inline suppression is annoying and waiting for a maintainer to suppress it through UI isn't an option. On the other hand only severe levels like errors should fail the build, and these are typically issues to address.

Btw. the *sizeof()* usage in the line above really looks Suspicious. :-)

**Comment by thetic** [comment]
Gotcha, thank you for the context. I think you're right, then; we should keep CodeQL.

**Comment by offa** [comment]
This shouldn't block or otherwise stop this PR of course! :+1:

File: src/CppUTest/TestMemoryAllocator.cpp [hunk 1]
```diff
@@ -418,7 +418,7 @@
 
 void MemoryAccountant::destroyAccountantAllocationNode(MemoryAccountantAllocationNode* node) const
 {
-    allocator_->free_memory((char*) node, sizeof(node), __FILE__, __LINE__);
+    allocator_->free_memory((char*) node, sizeof(node), __FILE__, __LINE__); // NOLINT
```
[the remark this answers is not in the document]
Comment (offa) [reply src/CppUTest/TestMemoryAllocator.cpp:421]: Not related to this PR, but is this really correct @basvodde !?

[the remark this answers is not in the document]
Comment (basvodde) [reply src/CppUTest/TestMemoryAllocator.cpp:421]: You mean, the sizeof(node) I assume. It doesn't really look correct to me at first sight. I guess it hasn't caused a problem as the size is not actually used in free_memory allocator except for printing ?

**Comment by thetic** [comment]
> Ok, many small fixes. I'll merge it even though I see some builds not ready yet.

Kill any jobs you'd like on appveyor. Nothing will pass until #1702 merges.

**Comment by basvodde** [comment]
Merged before I read this :)

**Pull request merged**